src/features/CanvasEditor.tsx

```diff
@@ -32,144 +43,66 @@ const defaultDefinition = {
-  const [selectedComponent, setSelectedComponent] = useState({});
-  const [hoveredComponent, setHoveredComponent] = useState(null);
-  const [currentPageID, setCurrentPageID] = useState(1);
-  const [zoomLevel, setZoomLevel] = useState(1.0);
-  const [rendering, setRendering] = useState(false);
+/**
+ * CanvasEditor
+ *
+ * Description: This return entire canvas and component library
+ */
 
+const CanvasEditor = () => {
+  const canvasContainerRef = useRef<HTMLInputElement>(null);
+  /**
+   * set isDragging when component Drag
+   */
+  const [isDragging, setIsDragging] = useState(false);
+  /**
+   * set the current Page
+   */
+  const [currentPage, setCurrentPage] = useState<string>(defaultPageId);
+  /**
+   * set the app Definition of the project
+   */
+  const [appDefinition, setAppDefinition] =
+    useState<IAppDefination>(defaultDefinition);
+  /**
+   * set rendering till page gets loaded for first time
+   *  - this is now working  but incase for the first time if
+   *    component is does not drag the we have to add it
+   */
+  // const [rendering, setRendering] = useState(stubTrue);
+
+  const zoomLevel = 1;
+
+  /**
+   * Wait for 500 seconds to page get loaded
+   * - this is now working  but incase for the first time if
+   *    component is does not drag the we have to add it
+   */
   useEffect(() => {
-    const run = debounce(() => {
-      setRendering(true);
-    }, 500);
-    run();
+    // const run = debounce(() => {
+    //   setRendering(true);
+    // }, 500);
+    // run();
   }, []);
 
-  const usePrevious = value => {
-    const ref = useRef();
-    useEffect(() => {
-      ref.current = value;
-    }, [value]);
-    return ref.current;
-  };
-
-  // useEffect(() => {
-  //   console.log("CanvasContainerRef", canvasContainerRef);
-  // });
-
-  const previous = usePrevious(appDefinition);
-
-  // useEffect(() => {
-  //   console.log("previeosu ststate", previous);
-  // }, [previous]);
-
-  const appDefinitionChanged = (newDefinition, opts = {}) => {
-    let currentPageId = currentPageID;
-    if (isEqual(appDefinition, newDefinition)) return;
-    // if (config.ENABLE_MULTIPLAYER_EDITING && !opts.skipYmapUpdate) {
-    //   this.props.ymap?.set("appDef", {
-    //     newDefinition,
-    //     editingVersionId: this.props.editingVersion?.id,
-    //   });
-    // }
-
-    if (opts?.versionChanged) {
-      currentPageId = newDefinition.homePageId;
-
-      setAppDefinition({ ...newDefinition });
-      setCurrentPageID(currentPageId);
-      // this.setState(
-      //   {
-      //     isSaving: true,
-      //     currentPageId: currentPageId,
-      //     appDefinition: newDefinition,
-      //     appDefinitionLocalVersion: uuid(),
-      //   },
-      //   () => {
-      //     if (!opts.skipAutoSave) this.autoSave();
-      //     this.switchPage(currentPageId);
-      //   }
-      // );
-      return;
+  useEffect(() => {
+    if (!currentPage) {
```

Comment:
Can we please add comments what logic is specifying

Reply:
Currently, current Page is not used, but when will add multiple page that will be updated.